text-serializer-legacy/src/main/java/net/kyori/adventure/text/serializer/legacy/LegacyComponentSerializerImpl.java

```diff
@@ -234,7 +234,7 @@ private TextComponent extractUrl(final TextComponent component) {
     if (this.urlReplacementConfig == null) return component;
     final Component newComponent = component.replaceText(this.urlReplacementConfig);
     if (newComponent instanceof TextComponent) return (TextComponent) newComponent;
-    return TextComponent.ofChildren(newComponent);
+    return Component.text().append(newComponent).build();
```

Comment:
Is there a reason you didn't use the replacement recommended in the JD for `ofChildren`? If that issue exists here, what does that say for other users of the API? If it's not good enough for us, why should we ask others to migrate?

Reply:
Well, the annoying part about this methods and others like it is that they return `TextComponent` instead of `Component`. So in this situation I have to choice of using the new join method and then wrapping it in a text component (like I have done elsewhere) or just appending it as one child. In hindsight, this should be using an `instanceof` check before the wrapping is done as in a lot of cases this component will already be a `TextComponent`.